neuromation/cli/formatters/ftable.py

```diff
@@ -0,0 +1,120 @@
+from dataclasses import dataclass
+from enum import Enum
+from itertools import zip_longest
+from textwrap import wrap
+from typing import Any, Iterator, List, Optional, Sequence
+
+
+__all__ = ["table"]
+__version__ = "0.1"
+
+
+@dataclass(frozen=True)
+class ColumnWidth:
+    min: Optional[int] = None
+    max: Optional[int] = None
+    width: Optional[int] = None
+
+    def __post_init__(self) -> None:
+        if self.width and self.width < 0:
+            raise ValueError("Width must be positive integer")
+        if self.min and self.min < 0:
+            raise ValueError("Min must be positive integer")
+        if self.max and self.max < 0:
+            raise ValueError("Max must be positive integer")
+
+
+class Align(str, Enum):
+    CENTER = "center"
+    LEFT = "left"
+    RIGHT = "right"
+
+
+_align_to_format = {
+    Align.LEFT: str.ljust,
+    Align.RIGHT: str.rjust,
+    Align.CENTER: str.center,
+    None: str.ljust,
+}
+
+
+def table(
+    rows: Sequence[Sequence[str]],
+    widths: Sequence[ColumnWidth] = (),
+    aligns: Sequence[Align] = (),
+    max_width: Optional[int] = None,
+) -> Iterator[str]:
+
+    # Columns widths calculation
+    if len(widths) < len(rows[0]):
+        widths = tuple(widths) + tuple([ColumnWidth()]) * (len(rows[0]) - len(widths))
+    calc_widths: List[int] = []
+    for i, width in enumerate(widths):
+        max_cell_width: int = max(len(row[i]) for row in rows)
+        width_min = width.min if width.min else width.width
+        width_max = width.max if width.max else width.width
+
+        if (not width_min or width_min <= max_cell_width) and (
+            not width_max or width_max >= max_cell_width
+        ):
+            calc_widths.append(max_cell_width)
+        elif width_max:
+            calc_widths.append(width_max)
+        else:
+            calc_widths.append(max_cell_width)
+
+    # How many empty columns can be displayed
+    max_empty_columns = len(rows[0])
+    if max_width:
+        sum_width = 0
+        for i, column_width in enumerate(calc_widths):
+            sum_width += column_width
+            if sum_width > max_width:
```

Comment:
Does `max_width` include space between columns?

Reply:
you are right
fixing